llvm/include/llvm/IR/GlobalValue.h

```diff
@@ -578,18 +583,23 @@ class GlobalValue : public Constant {
                                          GlobalValue::LinkageTypes Linkage,
                                          StringRef FileName);
 
+private:
   /// Return the modified name for this global value suitable to be
   /// used as the key for a global lookup (e.g. profile or ThinLTO).
   std::string getGlobalIdentifier() const;
 
-  /// Declare a type to represent a global unique identifier for a global value.
-  /// This is a 64 bits hash that is used by PGO and ThinLTO to have a compact
-  /// unique way to identify a symbol.
-  using GUID = uint64_t;
-
   /// Return a 64-bit global unique ID constructed from global value name
   /// (i.e. returned by getGlobalIdentifier()).
-  static GUID getGUID(StringRef GlobalName);
+  static GUID getGUID(StringRef GlobalIdentifier);
```

Comment:
Where are the remaining callers of this function, besides from the new getGUIDAssumingExternalLinkage below?

Reply:
Good point -- there are no remaining users so these two functions are redundant now. I've removed the static `getGUID` now.